metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/AuthEnabledConnectorDatabaseService.java

```diff
@@ -0,0 +1,112 @@
+package com.netflix.metacat.common.server.connectors;
+
+import com.netflix.metacat.common.MetacatRequestContext;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.Pageable;
+import com.netflix.metacat.common.dto.Sort;
+import com.netflix.metacat.common.exception.MetacatUnAuthorizedException;
+import com.netflix.metacat.common.server.api.authorization.Authorization;
+import com.netflix.metacat.common.server.api.authorization.AuthorizationStatus;
+import com.netflix.metacat.common.server.connectors.model.DatabaseInfo;
+import com.netflix.metacat.common.server.util.MetacatContextManager;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Connector that authorizes requests based on the request context.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class AuthEnabledConnectorDatabaseService implements ConnectorDatabaseService {
```

Comment:
There is a `ConnectorBaseService` that the different types extend so having certain generalizable functionality that is needed across different types there is also an option if you see such an opportunity though up to you.